File: python/strategies/abstract_strategy.py
class AbstractStrategy:
    """The blueprint for all game strategies."""
    def get_info(self):
        return {"apiversion": "1", "author": "YourTeam", "color": "#888888", "head": "default", "tail": "default"}

    def on_game_start(self, game_state: dict):
        pass

    def on_game_move(self, game_state: dict) -> dict:
        raise NotImplementedError("You must implement the on_game_move method!")

    def on_game_end(self, game_state: dict):
        pass

File: typescript/strategies/AbstractStrategy.ts
import { GameState, InfoResponse } from '../types';

/**
 * The blueprint for all game strategies.
 */
export abstract class AbstractStrategy {
  /**
   * Get information about the Battlesnake
   * @returns InfoResponse containing bot details
   */
  public abstract get_info(): InfoResponse;

  /**
   * Called when a game starts
   * @param gameState The initial game state
   */
  public on_game_start(gameState: GameState): void {
    // Default implementation does nothing
  }

  /**
   * Called on every turn to determine the next move
   * @param gameState The current game state
   * @returns MoveResponse containing the chosen move
   */
  public abstract on_game_move(gameState: GameState): { move: string };

  /**
   * Called when a game ends
   * @param gameState The final game state
   */
  public on_game_end(gameState: GameState): void {
    // Default implementation does nothing
  }
}